fermenter-mda/src/main/java/org/technologybrewery/fermenter/mda/notification/NotificationService.java

```diff
@@ -69,56 +83,58 @@ static void addNotificationToPassedMap(String targetFile, Notification notificat
     }
 
     /**
-     * Merges any notifications from the current plugin execution into the session for storage across plugin invocations.
+     * Writes any encountered notifications to a file for use later.  These are written to the path defined by
+     * NOTIFICATION_DIRECTORY_PATH.
      */
-    public synchronized void mergeNotificationsIntoSessionForCrossProjectStorage() {
+    public void recordNotifications() {
+        int manualActionCount = 0;
+        File projectParentFile = new File(NOTIFICATION_DIRECTORY_PATH);
         Map<String, Map<String, Notification>> collectorNotifications = NotificationCollector.getNotifications();
-        Map<String, Map<String, Notification>> sessionNotifications = getNotificationsFromSession();
-
-        synchronized (this) {
-            if (sessionNotifications == null) {
-                sessionNotifications = collectorNotifications;
-                Properties userProperties = session.getUserProperties();
-                userProperties.put(NOTIFICATIONS, sessionNotifications);
-            } else {
-                for (Map.Entry<String, Map<String, Notification>> entry : collectorNotifications.entrySet()) {
-                    String fileName = entry.getKey();
-                    Map<String, Notification> notificationsForFile = entry.getValue();
-                    for (Map.Entry<String, Notification> subMapEntry : notificationsForFile.entrySet()) {
-                        addNotificationToPassedMap(fileName, subMapEntry.getValue(), sessionNotifications);
-                    }
+        for (Map.Entry<String, Map<String, Notification>> entry : collectorNotifications.entrySet()) {
+            String fileName = entry.getKey();
+            Map<String, Notification> notificationsForFile = entry.getValue();
+            int i = 0;
+            for (Map.Entry<String, Notification> subMapEntry : notificationsForFile.entrySet()) {
+                File outputFile = new File(projectParentFile, FilenameUtils.getName(fileName + "-" + i++ + ".txt"));
+                try {
+                    FileUtils.forceMkdir(projectParentFile);
+                    FileUtils.write(outputFile, subMapEntry.getValue().getNotificationAsString(), Charset.defaultCharset());
+                    manualActionCount++;
+                } catch (IOException e) {
+                    throw new GenerationException("Could not write manual action notification to disk!", e);
                 }
             }
+
+            if (manualActionCount > 0) {
```

Comment:
A: We are baking in the notion that "notifications" == "manual actions".  Maybe/probably fine for incremental addition, but I could see this being more universal, e.g. profile deprecation notifications.

Reply:
Yeah - agree.  Debated this but didn't have a good example of what wouldn't be a manual action that we are outputting.  Easy to refactor to a more general concept if we want to, however.